src/dagmc.cpp

```diff
@@ -0,0 +1,406 @@
+#include "openmc/dagmc.h"
+
+#include "openmc/cell.h"
+#include "openmc/constants.h"
+#include "openmc/error.h"
+#include "openmc/file_utils.h"
+#include "openmc/geometry.h"
+#include "openmc/geometry_aux.h"
+#include "openmc/material.h"
+#include "openmc/string_utils.h"
+#include "openmc/settings.h"
+#include "openmc/surface.h"
+
+#ifdef DAGMC
+
+#include "uwuw.hpp"
+#include "dagmcmetadata.hpp"
+
+#endif
+
+#include <string>
+#include <sstream>
+#include <algorithm>
+#include <fstream>
+
+namespace openmc {
+
+#ifdef DAGMC
+const bool dagmc_enabled = true;
+#else
+const bool dagmc_enabled = false;
+#endif
+
+}
+
+#ifdef DAGMC
+
+namespace openmc {
+
+const std::string DAGMC_FILENAME = "dagmc.h5m";
+
+namespace simulation {
+
+moab::DagMC::RayHistory history;
+Direction last_dir;
+
+}
+
+namespace model {
+
+moab::DagMC* DAG;
+
+} // namespace model
+
+
+void check_dagmc_file() {
+  std::string filename = settings::path_input + DAGMC_FILENAME;
+  if (!file_exists(filename)) {
+    fatal_error("Geometry DAGMC file '" + filename + "' does not exist!");
+  }
+}
+
+bool get_uwuw_materials_xml(std::string& s) {
+  check_dagmc_file();
+  UWUW uwuw((settings::path_input + DAGMC_FILENAME).c_str());
+
+  std::stringstream ss;
+  bool uwuw_mats_present = false;
+  if (uwuw.material_library.size() != 0) {
+    uwuw_mats_present = true;
+    // write header
+    ss << "<?xml version=\"1.0\"?>\n";
+    ss << "<materials>\n";
+    const auto& mat_lib = uwuw.material_library;
+    // write materials
+    for (auto mat : mat_lib) { ss << mat.second.openmc("atom"); }
+    // write footer
+    ss << "</materials>";
+    s = ss.str();
+  }
+
+  return uwuw_mats_present;
+}
+
+bool read_uwuw_materials(pugi::xml_document& doc) {
+  std::string s;
+  bool found_uwuw_mats = get_uwuw_materials_xml(s);
+  if (found_uwuw_mats) {
+    pugi::xml_parse_result result = doc.load_string(s.c_str());
```

Comment:
Did you mean to ignore the result on line 89?  If so maybe a comment here could help future readers of the code.